src/link/output.c

```diff
@@ -30,6 +31,15 @@ FILE *overlayFile;
 FILE *symFile;
 FILE *mapFile;
 
+// The map file needs its summary output first, but the information in the summary is accumulated
+// while outputting the individual banks. So the banks are buffered here, then the summary is
+// output, then the accumulated banks are output all at once.
+struct {
+	size_t length;
+	size_t capacity;
+	char *data;
+} mapBuffer;
```

Comment:
Rather than incurring the cost of repeated allocations (which I'd expect to be O(n)), wouldn't it be better to loop over the data twice (first to collect the summary, then to output the details)?

Reply:
The allocations double the capacity each time, so it's O(log n).

If you'd still prefer this to be done in two passes, I can make that change.

Reply:
I feel like it'd remove a lot of string manipulation, thus being faster and simpler

Reply:
Thanks, that *is* simpler.